### PR TITLE
Add a fake package that generates a link to the Android javadocs in the dartdoc output

### LINKED ISSUE #7963: Link to javadocs from docs.flutter.io (and maybe in search too?)
https://github.com/flutter/flutter/pull/7935 added javadocs to docs.flutter.io:
https://docs.flutter.io/javadoc/io/flutter/view/FlutterView.html

But I we don't currently link to them from our front page.

The way we create the index.html for docs.flutter.io is a bit odd however:
https://github.com/flutter/flutter/blob/master/dev/tools/dartdoc.dart#L106

FYI @LarkAscending @jason-simmons

### PR DESCRIPTION
**Pull request by jason-simmons**
Fixes https://github.com/flutter/flutter/issues/7963

### CODE REVIEW

**Comment by jason-simmons** [comment]
@LarkAscending @sethladd

**Comment by Hixie** [comment]
clever!
LGTM

File: dev/docs/platform_integration/lib/android.dart [hunk 1]
```diff
@@ -0,0 +1,2 @@
+/// [Flutter platform integration APIs for Android.](https://docs.flutter.io/javadoc)
```
Comment (abarth) [inline dev/docs/platform_integration/lib/android.dart]: Add trailing `/` to avoid one network round-trip

File: dev/tools/dartdoc.dart [hunk 2]
```diff
@@ -134,6 +139,10 @@ void addHtmlBaseToIndex() {
   String indexContents = indexFile.readAsStringSync();
   indexContents = indexContents.replaceFirst('</title>\n',
     '</title>\n  <base href="./flutter/">\n');
+  indexContents = indexContents.replaceAll(
+    'href="Android/Android-library.html"',
+    'href="https://docs.flutter.io/javadoc"'
```
Comment (abarth) [inline dev/tools/dartdoc.dart]: Also needs `/`

**Comment by abarth** [comment]
LGTM

**Pull request merged**